### PR TITLE
Add support for RHEL9/CentOS9 as bastion

### PR DESCRIPTION
**Pull request by cs-zhang**
#408 
Test with RHEL9 and terraform v1.2.5
Signed-off-by: CS Zhang <zhangcho@us.ibm.com>

### CODE REVIEW

**Review by sajauddin — DISMISSED** [review]
/lgtm

**Comment by aishwaryabk** [comment]
@cs-zhang Could we mention in the documentation that it will only work with terraform version> 1.2.0?

File: modules/1_prepare/prepare.tf [hunk 1]
```diff
@@ -299,7 +299,12 @@ resource "null_resource" "enable_repos" {
 if ( [[ -z "${var.rhel_subscription_username}" ]] || [[ "${var.rhel_subscription_username}" == "<subscription-id>" ]] ) && [[ -z "${var.rhel_subscription_org}" ]]; then
   sudo yum install -y epel-release
 else
-  sudo subscription-manager repos --enable ${var.ansible_repo_name}
+  os_ver=$(cat /etc/os-release | egrep "^VERSION_ID=" | awk -F'"' '{print $2}')
+  if [[ $os_ver != "9"* ]]; then
+    sudo subscription-manager repos --enable ${var.ansible_repo_name}
+  else
+    sudo yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
```
Comment (yussufsh) [inline modules/1_prepare/prepare.tf:306]: Is this related to RHEL or only specific to CentOS? If the answer is yes then you need to check this in above if condition.

File: modules/1_prepare/prepare.tf [hunk 2]
```diff
@@ -322,7 +327,7 @@ resource "null_resource" "bastion_packages" {
   provisioner "remote-exec" {
     inline = [
       "#sudo yum update -y --skip-broken",
-      "sudo yum install -y wget jq git net-tools vim python3 tar"
+      "sudo yum install -y wget jq git net-tools vim python3 tar grub2-tools-extra"
```
Comment (yussufsh) [inline modules/1_prepare/prepare.tf:330]: What is the purpose of this package with context to this PR? Suggest another PR or commit with the details.

File: modules/5_install/install.tf [hunk 3]
```diff
@@ -207,6 +207,7 @@ resource "null_resource" "config" {
   provisioner "remote-exec" {
     inline = [
       "sed -i \"/^helper:.*/a \\ \\ networkifacename: $(ip r | grep \"${var.cidr} dev\" | awk '{print $3}')\" ocp4-helpernode/helpernode_vars.yaml",
+      "sed -i 's/when: ansible_distribution_major_version == \"8\"/when: ansible_distribution_major_version >= \"8\"/g' ocp4-helpernode/tasks/set_facts_.yaml",
```
Comment (yussufsh) [inline modules/5_install/install.tf:210]: helpernode owner is not allowing this change? should we work on the fork for our team? Let me know your thoughts.

**Review by yussufsh — CHANGES_REQUESTED** [review]
minor doubts.

**Comment by yussufsh** [comment]
> @cs-zhang Could we mention in the documentation that it will only work with terraform version> 1.2.0?

Agreed. We should mention it as a particular case for RHEL9. Also, I am okay with bumping up the versions.tf entries.

File: modules/5_install/install.tf (hunk 3 above)
Comment (aishwaryabk) [reply modules/5_install/install.tf:210]: I believe this change is already merged in the helpernode repo https://github.com/redhat-cop/ocp4-helpernode/pull/291

Comment (yussufsh) [reply modules/5_install/install.tf:210]: Perhaps not promoted to the main branch.

Comment (aishwaryabk) [reply modules/5_install/install.tf:210]: Yes

File: modules/1_prepare/prepare.tf (hunk 1 above)
Comment (cs-zhang) [reply modules/1_prepare/prepare.tf:306]: this is for RHEL 9 only.  For CentOS 9, it uses the same way as CentOS 8 to install epel repo.

File: modules/1_prepare/prepare.tf (hunk 2 above)
Comment (cs-zhang) [reply modules/1_prepare/prepare.tf:330]: this is to install grub2-mknetdir, which ocp4-helpernode will use.

File: modules/5_install/install.tf (hunk 3 above)
Comment (cs-zhang) [reply modules/5_install/install.tf:210]: yes, it is not in main branch yet, it still works even if it merged to main branch.

**Comment by yussufsh** [comment]
> > @cs-zhang Could we mention in the documentation that it will only work with terraform version> 1.2.0?
> 
> Agreed. We should mention it as a particular case for RHEL9. Also, I am okay with bumping up the versions.tf entries.

@cs-zhang we should change the minimum version of Terraform now. I have also noticed at https://cloud.ibm.com/docs/ibm-cloud-provider-for-terraform?topic=ibm-cloud-provider-for-terraform-getting-started ```The IBM Cloud Provider plug-in for Terraform currently supports Terraform stable version 1.1.x```

File: docs/automation_host_prereqs.md [hunk 4]
```diff
@@ -21,7 +21,7 @@ Install the following packages on the automation host. Select the appropriate in
 
 ### Terraform
 
-**Terraform**: Please open the [link](https://www.terraform.io/downloads) for downloading the latest Terraform. For validating the version run `terraform version` command after install.
+**Terraform**: Please open the [link](https://www.terraform.io/downloads) for downloading the latest Terraform. For validating the version run `terraform version` command after install. Terraform version 1.2.0 and above is required for CentOS9/RHEL9.
```
Comment (yussufsh) [inline docs/automation_host_prereqs.md:24]: No need to mention rhel9 as it is applicable for all versions now.

File: modules/1_prepare/prepare.tf (hunk 2 above)
Comment (cs-zhang) [reply modules/1_prepare/prepare.tf:330]: This is required for CentOS9/RHEL9.

**Review by yussufsh — APPROVED** [review]
